src/lightning/pytorch/trainer/trainer.py

```diff
@@ -749,6 +747,9 @@ def test(
             model = _maybe_unwrap_optimized(model)
             self.strategy._lightning_module = model
         _verify_strategy_supports_compile(self.lightning_module, self.strategy)
+        self.state.fn = TrainerFn.TESTING
+        self.state.status = TrainerStatus.RUNNING
+        self.testing = True
```

Comment:
When we `.fit()` the trainer state saved by spawn is still `FITTING`, even though it will be done right after.

This means that the main process is recovered with the "wrong" trainer state.

Moving this makes sure that the state is correct before spawning. The added test will fail without this